features/dd-sdk-android-trace/src/test/kotlin/com/datadog/opentracing/PendingTraceTest.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.opentracing
+
+import com.datadog.android.utils.forge.Configurator
+import com.datadog.trace.api.sampling.PrioritySampling
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.whenever
+import org.mockito.quality.Strictness
+import java.math.BigInteger
+import java.util.ConcurrentModificationException
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(value = Configurator::class)
+class PendingTraceTest {
+
+    @Mock
+    lateinit var mockTracer: DDTracer
+
+    lateinit var fakeTraceId: BigInteger
+
+    @Test
+    fun `PendingTrace is thread safe`(forge: Forge) {
+        whenever(mockTracer.partialFlushMinSpans) doReturn 1
+        fakeTraceId = BigInteger.valueOf(forge.aLong())
+        val pendingTrace = PendingTrace(mockTracer, fakeTraceId)
+        val rootSpan = forge.fakeSpan(pendingTrace, mockTracer, fakeTraceId, BigInteger.ZERO)
+        val countDownLatch = CountDownLatch(2)
```

Comment:
why do we have 2 for the latch, but starting 4 threads which are decrementing the latch? shouldn't the numbers be the same